Source/RunActivity/Viewer3D/Viewer.cs

```diff
@@ -1364,29 +1368,38 @@ void HandleUserInput(ElapsedTime elapsedTime)
 
             if (Camera is CabCamera && (PlayerLocomotiveViewer as MSTSLocomotiveViewer)._hasCabRenderer)
             {
-                if (UserInput.IsMouseLeftButtonPressed)
+                if (UserInput.IsMouseLeftButtonPressed || UserInput.IsMouseWheelChanged)
                 {
                     var cabRenderer = (PlayerLocomotiveViewer as MSTSLocomotiveViewer)._CabRenderer;
                     foreach (var controlRenderer in cabRenderer.ControlMap.Values)
                     {
-                        if ((Camera as CabCamera).SideLocation == controlRenderer.Control.CabViewpoint && controlRenderer is ICabViewMouseControlRenderer mouseRenderer && mouseRenderer.IsMouseWithin())
+                        if ((Camera as CabCamera).SideLocation == controlRenderer.Control.CabViewpoint && controlRenderer is ICabViewMouseControlRenderer mouseRenderer)
                         {
-                            if ((controlRenderer.Control.Screens == null || controlRenderer.Control.Screens[0] == "all"))
+                            if (mouseRenderer.IsMouseWithin())
                             {
-                                MouseChangingControl = mouseRenderer;
-                                break;
-                            }
-                            else
-                            {
-                                foreach (var screen in controlRenderer.Control.Screens)
+                                var UserCommandControlTypes = (PlayerLocomotiveViewer as MSTSLocomotiveViewer).UserCommandControlTypes;
+                                if (UserCommandControlTypes.ContainsKey(controlRenderer.Control.ControlType.Type) && UserInput.IsMouseWheelChanged)
                                 {
-                                    if (cabRenderer.ActiveScreen[controlRenderer.Control.Display] == screen)
+                                    continue;
```

Comment:
I found that by changing this line of code out for
`MouseChangingControl = mouseRenderer; break;`
and reverting the changes to MSTSLocomotiveViewer.HandleUserInput line 339, the mouse scrolling became much more responsive and predictable. Because this method uses the same behavior as moving the mouse around, it doesn't need any of the input buffering, it doesn't freak out when trying to move multiple controls, spinning the mousewheel faster causes the control to move faster. Overall a vastly better experience that doesn't need to rely on workarounds.

I also think the resulting scroll direction behavior is more intuitive, as scrolling down moves controls down/left and scrolling up moves controls up/right, regardless of which direction is increasing/decreasing the control. While this might seem less consistent (sometimes scrolling up increases the brakes, sometimes it decreases the brakes, depends on the cab view) it makes for a better connection between the motion of my hand and the motion of the levers in game.

The only problem with this strategy that makes it worse than the current simulated button press strategy is that notched controls may take more than one click of the scroll wheel to advance by one notch.

Reply:
hi Phillip, thanks for your comments. I wasn't already not that satisfied with my code changes. This should have been implemented right from the start of MSTS/OR. But it might be there was no scroll wheel back then. 
I'll have a look at your comments once I'm finished with the review comments for my download PR.

Reply:
hi Philip, 
I'm afraid this is all too difficult for me. As I said I was already not that satisfied with my changes. 
I'm thinking of undoing these changes (except the ctrl-F5). Or would you like to take over this PR? Because it's still a nice thing to have, mouse wheel support. It's not that I'm dissatisfied with your comments, on the contrary they look good to me.
Please let me know what you think,
regards, Siebren.

Reply:
I think it's a good idea to split the Ctrl + F5 and mouse wheel into two different PRs. They are both useful but aren't so closely connected that they need to be developed at the same time. Since the control hitboxes are already implemented, that PR could be quickly approved without the distraction of the mouse wheel feature.

I can investigate the scroll wheel feature in my own time, since I think I know some good ways to pull that off.